src/endpoints/market/v1.ts

```diff
@@ -0,0 +1,985 @@
+            page?: number
+            limit?: number
+            order?: 'asc' | 'desc'
+            sort?: 'template_id' | 'price'
+        },
+        extra_options?: {[key: string]: string}
+    ) {
+        const queryParts = {}
+
+        for (const [key, value] of Object.entries(options || {})) {
+            queryParts[key] = value as string
+        }
+        for (const [key, value] of Object.entries(extra_options || {})) {
+            queryParts[key] = value
+        }
+
+        const queryParams = Object.keys(queryParts).length
+            ? '?' + new URLSearchParams(queryParts).toString()
+            : ''
+
+        return this.client.call({
+            path: `/atomicmarket/v1/sales/templates${queryParams}`,
+            method: 'GET',
+            responseType: Market.GetSalesTemplatesResponse,
+        })
+    }
+
+    async get_auctions(
+        options?: {
+            state?: AuctionState[]
+            bidder?: NameType[]
+            participant?: NameType
+            hide_empty_auctions?: boolean
+            hide_templates_by_accounts?: NameType[]
+            max_assets?: UInt64Type
+            min_assets?: UInt64Type
+            show_seller_contracts?: boolean
+            contract_whitelist?: NameType[]
+            seller_blacklist?: NameType[]
+            buyer_blacklist?: NameType[]
+            asset_id?: UInt64Type[]
+            marketplace?: string[]
+            maker_marketplace?: string[]
+            taker_marketplace?: string[]
+            symbol?: string
+            account?: NameType[]
+            seller?: NameType[]
+            buyer?: NameType[]
+            min_price?: Float64Type
+            max_price?: Float64Type
+            min_template_mint?: UInt64Type
+            max_template_mint?: UInt64Type
+            collection_name?: NameType[]
+            schema_name?: NameType[]
+            template_id?: Int32Type[]
+            burned?: boolean
+            owner?: NameType[]
+            match?: string
+            search?: string
+            match_immutable_name?: string
+            match_mutable_name?: string
+            is_transferable?: boolean
+            is_burnable?: boolean
+            minter?: NameType[]
+            burner?: NameType[]
+            initial_receiver?: NameType[]
+            collection_blacklist?: NameType[]
+            collection_whitelist?: NameType[]
+            auction_id?: UInt64Type[]
+            ids?: UInt64Type[]
+            lower_bound?: string
+            upper_bound?: string
+            before?: number
+            after?: number
+            page?: number
+            limit?: number
+            order?: 'asc' | 'desc'
+            sort?:
+                | 'created'
+                | 'updated'
+                | 'ending'
+                | 'auction_id'
+                | 'price'
+                | 'template_mint'
+                | 'name'
+        },
+        extra_options?: {[key: string]: string}
```

Comment:
What are these extra options? They are not defined anywhere?

Reply:
From docs: `You can filter the result by specific asset / template data fields.You can add for example &template_data.rarity=common to only receive results which have an attribute "rarity" with the value "common". You can query specific asset data by using &immutable_data.rarity=common or &mutable_data.rarity=common .If you want to query a non text type you need to specify it explicitly (defaults to text type) like data:bool.foil=true or data:number.id=4 or data:text.rarity=common. Integers which are defined greater than 32 bit (eg 64 bit) in the schema need to be queried as text.`

Reply:
hmm so it might be difficult to come up with some better types then 🤔

Reply:
Yeah, i can think of other proper way to support this, we even have no idea what properties a `data` or `template_data` would have.